fili-core/src/main/java/com/yahoo/bard/webservice/data/config/dimension/DefaultDimensionConfig.java

```diff
@@ -0,0 +1,134 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.config.dimension;
+
+import com.yahoo.bard.webservice.data.config.names.DimensionName;
+import com.yahoo.bard.webservice.data.dimension.DimensionField;
+import com.yahoo.bard.webservice.data.dimension.KeyValueStore;
+import com.yahoo.bard.webservice.data.dimension.SearchProvider;
+
+import java.util.LinkedHashSet;
+import javax.validation.constraints.NotNull;
+
+
+/**
+ * A Dimension instance holds all of the information needed to construct a Dimension.
+ */
+public class DefaultDimensionConfig implements DimensionConfig {
```

Comment:
Remember, we have the `TypeAwareDimensionLoader` that does instances of in order to build `Dimensions` from `DimensionConfig`.